### PR TITLE
add orderingKey for pubSub sink

### PR DESCRIPTION
**Pull request by amitlopes-walmart**

### CODE REVIEW

File: packages/gcp/src/PubSubSink.ts [hunk 1]
```diff
@@ -33,10 +33,17 @@ interface IPayloadWithAttributes {
...
+}
+
+interface IPubSubTopicPayload {
+    payloadWithAttributes: IPayloadWithAttributes[];
+    messageOrdering: boolean;
 }
 
 export enum PubSubMetadata {
     Topic = "topic",
+    OrderingKey = "orderingKey",
```
Comment (butorine) [inline packages/gcp/src/PubSubSink.ts]: I would rename this to `Key = "orderingKey"` to be consistent with kafka nomenclature

File: packages/gcp/src/PubSubSink.ts [hunk 2]
```diff
@@ -70,15 +77,21 @@ export class PubSubSink
...
+                    payloadWithAttributes: [],
+                });
             }
 
             const formattedMsg = this.formatMessage(msg);
-            messagesByTopic.get(topic).push(formattedMsg);
+            const topic = messagesByTopic.get(topicName);
+
+            topic.payloadWithAttributes.push(formattedMsg);
+            topic.messageOrdering = !!formattedMsg.orderingKey;
```
Comment (butorine) [inline packages/gcp/src/PubSubSink.ts]: This should only be set if `topic.messageOrdering` is false.  If it's already true, then bypass.  This way if some messages contain a key, but some do not, then all will be published with an ordering key

File: packages/gcp/src/PubSubSink.ts [hunk 3]
```diff
@@ -33,10 +33,17 @@ interface IPayloadWithAttributes {
     payload: Buffer;
     attributes: Attributes;
     spanContext: SpanContext;
+    orderingKey: string;
+}
+
+interface IPubSubTopicPayload {
+    payloadWithAttributes: IPayloadWithAttributes[];
```
Comment (butorine) [inline packages/gcp/src/PubSubSink.ts]: Make this plural, please: `payloadsWithAttributes`

**Review by butorine — CHANGES_REQUESTED** [review]

File: packages/gcp/src/__test__/PubSubSink.test.ts [hunk 4]
```diff
@@ -139,12 +144,88 @@ describe("PubSubSink Tests", () => {
...
-            expect(mockPublishFn.mock.calls[idx][1][AttributeNames.contentType]).toBe(
+            expect(mockPublishFn.mock.calls[idx][0].attributes[AttributeNames.eventType]).toBe(
+                message.type
+            );
+            expect(mockPublishFn.mock.calls[idx][0].attributes[AttributeNames.contentType]).toBe(
                 pubSubPublisherConfigurationWithDefaultTopic.encoder.mimeType
             );
         });
     });
+    it("with orderingKey", async () => {
```
Comment (plameniv) [inline packages/gcp/src/__test__/PubSubSink.test.ts:158]: Can you make sure to have an empty line before each `it(...` line?

File: packages/gcp/src/__test__/PubSubSink.test.ts [hunk 5]
```diff
@@ -139,12 +144,88 @@ describe("PubSubSink Tests", () => {
...
+            expect(mockTopic.mock.calls[idx][1].messageOrdering).toBeTruthy();
+        });
+        expect(mockPublishFn).toBeCalledTimes(messagesWithTopic.length);
+        messagesWithTopic.forEach((message, idx) => {
+            expect(mockPublishFn.mock.calls[idx][0][PubSubMetadata.Key]).toBe(
+                message.payload.orderingKey
+            );
+        });
+    });
+    it("without orderingKey", async () => {
```
Comment (plameniv) [inline packages/gcp/src/__test__/PubSubSink.test.ts]: I think you can just move a couple of the expects which check for `no orderingKey` into the 2 already existing positive tests which don't use an ordering key.
And remove this `it(...` segment.

**Review by plameniv — APPROVED** [review]
LGTM!

**Pull request merged**